crates/core_simd/src/permute.rs

```diff
@@ -0,0 +1,28 @@
+macro_rules! impl_shuffle_lane {
+    { $name:ident, $fn:ident, $n:literal } => {
+        impl $name<$n> {
+            /// A const SIMD shuffle that takes 2 SIMD vectors and produces another vector, using
+            /// the indices in the const parameter. The first or "self" vector will have its lanes
+            /// indexed from 0, and the second vector will have its first lane indexed at $n.
+            ///
+            /// Some SIMD shuffle instructions can be quite slow, so avoiding them by loading data
+            /// into the desired patterns in advance is preferred, but shuffles are still faster
+            /// than storing and reloading from memory.
+            #[inline]
+            pub fn shuffle<const IDX: [u32; $n]>(self, second: Self) -> Self {
+                unsafe { crate::intrinsics::$fn(self, second, IDX) }
```

Comment:
How does rustc treat the indices? What happens if one goes out of range?  Is it UB that we somehow need to handle, or does rustc do something to it?  I vaguely recall it turns the index into `idx % lanes` but I'm not sure if that's actually true.

Reply:
Perhaps the behavior to the answer to those questions should be documented

Reply:
It errors, actually, and there are tests to assure it does so, see https://github.com/rust-lang/rust/pull/76768

Reply:
Documented this requirement, since I would not be surprised if people also expect this.